### PR TITLE
sqlite: move first read into a transaction

### PR DESCRIPTION
**Pull request by vrothberg**
According to an old upstream issue [1]: "If the first statement after BEGIN DEFERRED is a SELECT, then a read transaction is started. Subsequent write statements will upgrade the transaction to a write transaction if possible, or return SQLITE_BUSY."

So let's move the first SELECT under the same transaction as the table initialization.

[NO NEW TESTS NEEDED] as it's a hard to cause race.

[1] https://github.com/mattn/go-sqlite3/issues/274#issuecomment-1429054597

Fixes: #17859

#### Does this PR introduce a user-facing change?

```release-note
Fix a bug in the sqlite database backend where the first read access may fail.
```

### CODE REVIEW

**Comment by vrothberg** [comment]
@mheon @baude @edsantiago PTAL

**Comment by vrothberg** [comment]
> but to me it seems cleaner to have

Sure, that sounds good to me as well.  Waiting for feedback from @mheon, then I can wrap things up and repush.

**Comment by edsantiago** [comment]
Stepping back in with an observation: this would be a code flow that I find more natural:
```
    func InitTables() {
        schema := getSchema()
        if schema == CURRENT_SCHEMA
            return       // nothing to do. Don't even bother with the CREATE TABLE IF NOT EXIST

        if schema is not nil
            return migrateFromSchema(schema) // this is scary but deferred to future

        // fall through: database does not exist
        ...do the CREATE TABLE thing
    }
```
That fixes one concern I have: calling InitTables() after Migrate(). (that seems weird. Either Init, or Migrate, but never both). It could also speed things up a little by eliminating unnecessary CREATE TABLEs and transactions.

But it's very probable that I've missed something.

**Comment by vrothberg** [comment]
Done :heavy_check_mark: PTanotherL

**Comment by mheon** [comment]
LGTM
@containers/podman-maintainers PTAL

**Comment by edsantiago** [comment]
LGTM2. Thanks @vrothberg!

**Comment by vrothberg** [comment]
merge me

**Review by giuseppe — APPROVED** [review]
/lgtm

**Pull request merged**
